### PR TITLE
chore(build): Replace preset-node-5 and babel-preset-es2015-node4 w/ babel-preset-env

### PR DESCRIPTION
**Pull request by kaylieEB**
**Summary**
Replace [babel-preset-node 5](https://github.com/leebenson/babel-preset-node5) and [babel-preset-es2015-node4](https://github.com/jbach/babel-preset-es2015-node4) with babel-preset-env since the first two packages are deprecated and no longer maintained.

The downside is that `preset-env` is has the comprehensive list of plugins for each environment type, whereas previously packages only selected a few that we were using, so our build size increased. I'm curious to see why @yarnpkg/core 's thoughts are on this, and happy to close the PR if we'd rather stick with the old packages.

**Test plan**
Builds should build and tests should pass since there weren't any code changes 😃

### CODE REVIEW

**Comment by Daniel15** [comment]
@kaylieEB Hmm, I wonder why this is going to increase the .js build size by ~100 KB? Any ideas?

**Comment by kaylieEB** [comment]
@Daniel15 I'd assume so since js files still go through a [transpilation/build step via gulp](https://github.com/yarnpkg/yarn/blob/master/gulpfile.js#L36)?

**Comment by BYK** [comment]
@kaylieEB still WIP? What do you need to get unblocked?

**Comment by kaylieEB** [comment]
I guess node 6 part is actually good to go, depends if we want the update node 4 in one same pr. I'll get to it later this week, or otherwise feel free to take over the PR.

**Comment by BYK** [comment]
>I guess node 6 part is actually good to go, depends if we want the update node 4 in one same pr. I'll get to it later this week, or otherwise feel free to take over the PR.

Isn't it better to upgrade them together at once so there are no discrepancies or code/dependency duplication? I think we can wait a bit more until you have time to get to this :)

**Comment by BYK** [comment]
@kaylieEB ping! 😄

**Comment by kaylieEB** [comment]
@BYK ready for review!

File: __tests__/index.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ function expectAnErrorMessage(command: Promise<Array<?string>>, expectedMessage:
     .then(function() {
       throw new Error('the command did not fail');
     })
-    .catch(error => expect(error.message).toContain(expectedMessage));
```
Comment (arcanis) [inline __tests__/index.js]: What is this checking for?

**Comment by arcanis** [comment]
Will close/reopen to trigger a new CircleCI test, it seems one of them bugged :(

**Pull request merged**